### PR TITLE
Cg/replacer

### PR DESCRIPTION
**Pull request by cgestes**
Display replacement diffs!

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,40 +1,134 @@
...
 
 use clap::{App, Arg};
 use ignore::Walk;
+use std::path::Path;
+use std::fs::File;
+use std::io::prelude::*;
+use regex::Regex;
+use difference::{Changeset, Difference};
+
+#[allow(unused_must_use)]
```
Comment (dmerejkowsky) [inline src/main.rs]: maybe add a comment explaining why you need this ?

File: src/main.rs [hunk 2]
```diff
@@ -1,40 +1,134 @@
...
+fn term_show_diff(before: &str, after: &str) {
+   // Compare both texts, the third parameter defines the split level.
+    let Changeset { diffs, .. } = Changeset::new(before, after, "\n");
+
+    let mut t = term::stdout().unwrap();
+
+    for i in 0..diffs.len() {
```
Comment (dmerejkowsky) [inline src/main.rs]: for diff in &diffs ?

File: src/main.rs [hunk 3]
```diff
@@ -1,40 +1,134 @@
...
+
+
+pub struct Replacer {
+    re: Regex,
+    replacement: String,
+}
+
+impl Replacer {
+    pub fn new(pattern: &str, replacement: &str) -> Replacer {
+        let re = Regex::new(pattern).expect("aie aie aie");
```
Comment (dmerejkowsky) [inline src/main.rs]: no french

File: src/main.rs [hunk 4]
```diff
@@ -1,40 +1,134 @@
...
+        }
+    }
+
+    pub fn replace(&self, buf: &str) {
+        let after = self.re.replace_all(buf, self.replacement.as_str());
+
+        if buf != after {
+            term_show_diff(buf, &after)
+        }
+        // let re = Regex::new(r"^\d{4}-\d{2}-\d{2}$").unwrap();
```
Comment (dmerejkowsky) [inline src/main.rs]: dead code

File: src/main.rs [hunk 5]
```diff
@@ -1,40 +1,134 @@
...
+
+        // for m in greper.iter(data.as_bytes()) {
+        //     println!("match: {} - {}", m.start(), m.end())
+        // }
+    }
+}
+
+
+fn grep_file(replacer: &Replacer, file_name: &Path) {
```
Comment (dmerejkowsky) [inline src/main.rs]: bad name. replace_in_file ?

File: src/main.rs (hunk 1 above)
Comment (cgestes) [reply src/main.rs]: I .. dont know. It's copy paste.

it's not production ready code, I believe that we have to write our own replacer module.

File: src/main.rs (hunk 2 above)
Comment (cgestes) [reply src/main.rs]: I'll try. (copy pasted also). it has to be rewritten.

File: src/main.rs (hunk 3 above)
Comment (cgestes) [reply src/main.rs]: How should we handle error?

this panic... Can a constructor return a Result<Replacer> ?

File: src/main.rs (hunk 5 above)
Comment (cgestes) [reply src/main.rs]: true

**Comment by cgestes** [comment]
thank you.

This needs more work. (the creation of a replacer module)

File: src/main.rs (hunk 1 above)
Comment (cgestes) [reply src/main.rs]: is it fine with it?

The code has to be reworked anyway. (it has been moved to a replacer submodule)

File: src/main.rs (hunk 2 above)
Comment (cgestes) [reply src/main.rs]: fixed

File: src/main.rs (hunk 1 above)
Comment (cgestes) [reply src/main.rs]: fixed

File: src/main.rs (hunk 3 above)
Comment (cgestes) [reply src/main.rs]: fixed

File: src/main.rs (hunk 4 above)
Comment (cgestes) [reply src/main.rs]: fixed

File: src/main.rs (hunk 5 above)
Comment (cgestes) [reply src/main.rs]: fixed

**Comment by cgestes** [comment]
First working version. It does replace :)

**Comment by dmerejkowsky** [comment]
Would be nice to explain the switch to structopt in the commit message

**Comment by cgestes** [comment]
comment on structopt: done :)

**Pull request closed without merging**